File: LeetCode/Java_Solutions/169_Majority-Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int m = 0;
        int count = 0;
        for(int i = 0; i < nums.length; i++){
            if(count == 0){
                m = nums[i];
                count = 1;
            } else if(m == nums[i]){
                ++count;
            } else {
                --count;
            }
        }
        return m;
    }
}

File: LeetCode/C++_Solutions/169_Majority-Element.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = 0;
        int count = 0;
        for(int i = 0; i < nums.size(); i++){
            if(count == 0){
                majority = nums[i];
                count = 1;
            }else if(majority == nums[i]){
                ++count;
            }else {
                --count;
            }
        }
        return majority;
    }
};